runbotics/runbotics-orchestrator-ui/src/components/HistoryTable/HistoryTable.content.tsx

```diff
@@ -41,12 +43,19 @@ const HistoryTable = forwardRef<any, HistoryTableProps>(({
 }, ref) => {
     const dispatch = useDispatch();
     const tableRef = useRef<HTMLDivElement>(null);
-
     const processInstances = useSelector(processInstanceSelector);
     const { page: processInstancePage, loadingPage } = processInstances.all;
     const [panelInfoState, setPanelInfoState] = useState<PanelInfoState>({ show: false });
+    const [columns, setColumns] = React.useState<Column<IProcessInstance>[] >([]);
```

Comment:
It would be like
`const columns = useMemo(() => { const processColumns = await processInstanceColumns() }, [i18n.locale]);`

now you can skip useEffect

Reply:
now when `processInstanceColumns` is a function instead of simple object we should rephrase it into something like `getProcessInstanceColumns()`

Reply:
> It would be like `const columns = useMemo(() => { const processColumns = await processInstanceColumns() }, [i18n.locale]);`
> 
> now you can skip useEffect

Tried that and
- got error because of _await_ => added  _async_  inside _useMemo_
- even though i added return processColumns and checked with then(arr=>console.log(arr)) inside _useMemo_ i get array i need but outside when i try columns={columns} there's always a promise.

I couldn't figure out how to skip using useState, but managed to do this:

`const [columns, setColumns] = React.useState<Column<IProcessInstance>[] >([]);

  useMemo(async () => { 
      const processColumns = await processInstanceColumns(); 
      setColumns(processColumns);
  }, [i18n.language]);

and skip useEffect and prepareProcessInstanceColumns by that.

Reply:
maybe it's better to discuss this during a call

Reply:
renamed processInstanceColumns to getPprocessInstanceColumns